.github/workflows/build-and-publish-master.yaml

```diff
@@ -0,0 +1,35 @@
+name: Master Validation and Release
```

Comment:
Unfortunately can't really test this without merging & seeing what happens, ditto for the publish-release workflow that triggers on tags.